CreeDictionary/API/search/lookup.py

```diff
@@ -242,25 +141,49 @@ def _fetch_results(
     # todo: remind user "are you searching in cree/english?"
     # todo: allow inflected forms to be searched through English. (requires database migration
     #  since now EnglishKeywords are bound to lemmas)
-    for stemmed_keyword in stem_keywords(user_query):
+    for stemmed_keyword in stem_keywords(search_run.internal_query):
 
         lemma_ids = EnglishKeyword.objects.filter(text__iexact=stemmed_keyword).values(
             "lemma__id"
         )
 
         for wordform in Wordform.objects.filter(id__in=lemma_ids):
-            english_results.add(
-                EnglishResult(MatchedEnglish(user_query), wordform, Lemma(wordform))
-            )  # will become  (user_query, inflection.text, inflection.lemma)
+            search_run.add_result(
+                Result(wordform, target_language_keyword_match=stemmed_keyword)
+            )
 
         # explained above, preverbs should be presented
         for wordform in Wordform.objects.filter(
             Q(pos="IPV") | Q(inflectional_category="IPV") | Q(pos="PRON"),
             id__in=lemma_ids,
             as_is=True,
         ):
-            english_results.add(
-                EnglishResult(MatchedEnglish(user_query), wordform, Lemma(wordform))
-            )  # will become  (user_query, inflection.text, wordform)
+            search_run.add_result(Result(wordform, is_preverb_match=True))
+
+
+def filter_cw_wordforms(q: Iterable[Wordform]) -> Iterable[Wordform]:
+    """
+    return the wordforms that has definition from CW dictionary
+
+    :param q: an Iterable of Wordforms
```

Comment:
Is there a better name for `q`? I'm not totally sure it needs a better name since this function is so small, but it would be nice to have a better idea of what it is based on its name. Either way, this docstring explanation is nice 👍

Reply:
I did not write this code, I just moved it around, and this functionality really doesn’t belong in the dictionary server but in whatever process produces the source dictionary used by itwêwina; that said `queryset` is a better name than `q` here so I have renamed it.

Reply:
That's fair, I was pretty sure you had just moved it like you said. Thanks for making it clearer all the same!